dartagnan/src/main/java/com/dat3m/dartagnan/expression/op/CmpOp.java

```diff
@@ -50,8 +50,6 @@ public boolean combine(BigInteger a, BigInteger b){
         };
     }
 
-    // Due to constant propagation, and the lack of a proper type system
-    // we can end up with comparisons like "False == 1", thus the following two methods
     public boolean combine(boolean a, BigInteger b){
         return switch (this) {
             case EQ -> ((b.compareTo(BigInteger.ONE) == 0) == a);
```

Comment:
We should either fully remove both methods (if not use) or keep the comment.

Reply:
Rathen than fully removing the visitor, I would add a `Precondition.verify()` to check that the types match (just to be sure removing this does not backfire us if we start using the visitor again before rewritting it), remove the calls to `combine()` and then get rid of those methods.